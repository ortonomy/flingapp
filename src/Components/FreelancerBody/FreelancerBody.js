import React, { Component } from 'react';
import FreelancerBanner from '../FreelancerBanner/FreelancerBanner';
import FreelancerInfo from '../FreelancerInfo/FreelancerInfo';
import body from './FreelancerBody.module.css';

class FreelancerBody extends Component {
  render() {
    return (
      <div className={body.FreelancerBody}>
        <FreelancerBanner name='Jorge Brock' title='Chief Editor' status='active' />
        <FreelancerInfo />
      </div>
    )
  }
}

export default FreelancerBody;
