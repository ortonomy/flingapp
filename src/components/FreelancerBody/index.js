import React, { Component } from 'react';
import FreelancerBanner from '../FreelancerBanner';
import FreelancerInfo from '../FreelancerInfo';
import body from './FreelancerBody.module.css';

class FreelancerBody extends Component {
  render() {
    if (!this.props.loading) {
    return (
      <div className={body.FreelancerBody}>
        <FreelancerBanner name={this.props.freelancer.flFirstName + ' ' + this.props.freelancer.flLastName} title='Chief Editor' status='active' />
        <FreelancerInfo freelancer={this.props.freelancer} />
      </div>
    )
    }
    else {
      return (
        <div></div>
      )
    }
  }
}

export default FreelancerBody;
