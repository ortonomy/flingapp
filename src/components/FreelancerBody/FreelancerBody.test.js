import React from 'react';
import FreelancerBody from '../FreelancerBody';

const freelancer = {
  flFirstName: 'example',
  flLastName: 'freelancer'
}

it('renders without crashing', () => {
  const wrapper = shallow(<FreelancerBody freelancer={freelancer} />);
  expect(wrapper).toMatchSnapshot();
});
