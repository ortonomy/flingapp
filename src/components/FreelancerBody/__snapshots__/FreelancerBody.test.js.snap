// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FreelancerBody
    freelancer={
        Object {
            "flFirstName": "example",
            "flLastName": "freelancer",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FreelancerBanner
          name="example freelancer"
          status="active"
          title="Chief Editor"
/>,
        <FreelancerInfo
          freelancer={
                    Object {
                              "flFirstName": "example",
                              "flLastName": "freelancer",
                            }
          }
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "example freelancer",
          "status": "active",
          "title": "Chief Editor",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "freelancer": Object {
            "flFirstName": "example",
            "flLastName": "freelancer",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FreelancerBanner
            name="example freelancer"
            status="active"
            title="Chief Editor"
/>,
          <FreelancerInfo
            freelancer={
                        Object {
                                    "flFirstName": "example",
                                    "flLastName": "freelancer",
                                  }
            }
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "example freelancer",
            "status": "active",
            "title": "Chief Editor",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "freelancer": Object {
              "flFirstName": "example",
              "flLastName": "freelancer",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
